applications/fishing-map/src/features/timebar/Timebar.tsx

```diff
@@ -39,9 +53,93 @@ const TimebarWrapper = () => {
     },
     [setBookmark]
   )
+  const mapInstance = useMapboxInstance()
+  const tilesLoading = useTilesState(mapInstance).loading
+  const currentTimeChunkId = useCurrentTimeChunkId()
+  const mapStyle = useMapStyle()
 
-  if (!start || !end) return null
+  const [stackedActivity, setStackedActivity] = useState<any>()
+  useEffect(() => {
```

Comment:
I'd move what the effect does to a external method, for me it makes the component harder to read so what about creating something like `getStackedActivity()` and use it as: 
```
if (
   mapInstance ||
   mapStyle ||
   !tilesLoading ||
   timebarVisualisation === TimebarVisualisations.Heatmap
) {
  setStackedActivity(getStackedActivity(valuesByFrame))
}
```